pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -1979,11 +1981,27 @@ impl Timeline {
                 info!("no layers to compact with gc: no historic layers below gc_cutoff, gc_cutoff={}", gc_cutoff);
                 return Ok(());
             };
+            let Some(min_layer_lsn) = layers
+                .iter_historic_layers()
+                .filter(|desc| {
+                    if let Some(compact_above_lsn) = options.compact_above_lsn {
+                        desc.get_lsn_range().end > compact_above_lsn // strictly larger than compact_above_lsn
+                    } else {
+                        true
+                    }
+                })
+                .map(|desc| desc.get_lsn_range().start)
+                .min()
+            else {
+                info!("no layers to compact with gc: no historic layers above compact_above_lsn, compact_above_lsn={}", options.compact_above_lsn.unwrap_or_default());
+                return Ok(());
+            };
             // Then, pick all the layers that are below the max_layer_lsn. This is to ensure we can pick all single-key
             // layers to compact.
             let mut rewrite_layers = Vec::new();
             for desc in layers.iter_historic_layers() {
                 if desc.get_lsn_range().end <= max_layer_lsn
+                    && desc.get_lsn_range().start >= min_layer_lsn
                     && overlaps_with(&desc.get_key_range(), &compaction_key_range)
```

Comment:
This comment `// Then, pick all the layers` is also outdated with this new change.
I suggest you merge it into the updated comment I'm asking for above.